### PR TITLE
fix: only add default color once

### PR DESCRIPTION
**Pull request by pipiiiiii**
### Types

- [x] 🐛 Bug Fixes

### <samp>🤖 Generated by Copilot at 588e562</samp>

* Fix a bug that caused wrong token colors in monaco editor by checking for the scope property of the theme settings ([link](https://github.com/opensumi/core/pull/2753/files?diff=unified&w=0#diff-3448736c3c31a91f63f3a686023016388d5c73261dcb1768627c918421304239L670-R671))
* Add a scope property with an empty array to the default theme settings object to match the expected format for the `fillDefaults` function ([link](https://github.com/opensumi/core/pull/2753/files?diff=unified&w=0#diff-3448736c3c31a91f63f3a686023016388d5c73261dcb1768627c918421304239R681))

Issue: https://github.com/opensumi/core/issues/2686

每次初始化 theme 时，会修改 `themeSetting`引用，导致 https://github.com/opensumi/core/blob/main/packages/theme/src/browser/theme-data.ts#L576
这里 `index` 会不一致，每次颜色会错位一个
有因为`scope` 会被补充为 `['']`，所以初始化的判断逻辑改一下就可以了

### <samp>🤖 Generated by Copilot at 588e562</samp>

Fixed a bug in the workbench theme service that affected the monaco editor. Improved the default theme settings with a scope property.

### CODE REVIEW

File: packages/theme/src/browser/workbench.theme.service.ts [hunk 1]
```diff
@@ -667,7 +667,8 @@ class Theme implements ITheme {
   // 将encodedTokensColors转为monaco可用的形式
   private patchTokenColors() {
     // 当默认颜色不在settings当中时，此处不能使用之前那种直接给encodedTokenColors赋值的做法，会导致monaco使用时颜色错位（theia的bug
-    if (this.themeData.themeSettings.filter((setting) => !setting.scope).length === 0) {
+    // scope 会在后续的流程中被补充为 [''] 所以直接进行初始化，防止判断有误
+    if (this.themeData.themeSettings.filter((setting) => JSON.stringify(setting.scope) === '[""]').length === 0) {
```
Comment (Ricbet) [inline packages/theme/src/browser/workbench.theme.service.ts:671]: 为什么不直接判断 scope.length === 1 & scope[0] === '' 呢

Comment (pipiiiiii) [reply packages/theme/src/browser/workbench.theme.service.ts:671]: 因为 scope 有可能是字符串，不是数组，这两个判断不行，一定要判断是 scope === [''] 这个才行，但是这种 tslint 会报错，就用了 stringify

Comment (Ricbet) [reply packages/theme/src/browser/workbench.theme.service.ts:671]: 现在这种判断方式有点 hack

Comment (pipiiiiii) [reply packages/theme/src/browser/workbench.theme.service.ts]: 想了下，也可以，改了

File: packages/theme/src/browser/workbench.theme.service.ts [hunk 2]
```diff
@@ -667,7 +667,8 @@ class Theme implements ITheme {
   // 将encodedTokensColors转为monaco可用的形式
   private patchTokenColors() {
     // 当默认颜色不在settings当中时，此处不能使用之前那种直接给encodedTokenColors赋值的做法，会导致monaco使用时颜色错位（theia的bug
```
Comment (erha19) [inline packages/theme/src/browser/workbench.theme.service.ts:669]: 这里的注释跟下面优化到一段吧

**Review by Ricbet — APPROVED** [review]

**Review by Ricbet — APPROVED** [review]

**Pull request merged**